From 410f15031197e5a252afed9a1c8b85c4cee136b7 Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Wed, 31 Jan 2018 13:55:50 +0000
Subject: [PATCH] CGTS-8837 cinder volume cannot be deleted after swact

Using cinder::host causes a deprecation warning in puppet. After
switching to cinder::backend_host and locking/unlocking controller
followed by swact cinder volumes cannot be deleted anymore. This
happens because CONF.host is still used by cinder services (not
entirey replaced with CONF.backend_host).

Set cinder config DEFAULT/host to cinder::host if defined or
cinder::backend_host otherwise.
---
 manifests/init.pp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index bf23eaa..3ad82e4 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -472,6 +472,12 @@ instead.")
     $default_availability_zone_real = $default_availability_zone
   }
 
+  if is_service_default($host) {
+    $host_real = $backend_host
+  } else {
+    $host_real = $host
+  }
+  
   cinder_config {
     'DEFAULT/report_interval':                  value => $report_interval;
     'DEFAULT/service_down_time':                value => $service_down_time;
@@ -480,7 +486,7 @@ instead.")
     'DEFAULT/default_availability_zone':        value => $default_availability_zone_real;
     'DEFAULT/allow_availability_zone_fallback': value => $allow_availability_zone_fallback;
     'DEFAULT/image_conversion_dir':             value => $image_conversion_dir;
-    'DEFAULT/host':                             value => $host;
+    'DEFAULT/host':                             value => $host_real;
     'DEFAULT/backend_host':                     value => $backend_host;
   }
 
-- 
1.8.3.1

